From d03ca11d9b64935af52929fbcf887b0e65c45521 Mon Sep 17 00:00:00 2001
From: Karthik D A <karthikda@maxlinear.com>
Date: Sun, 13 Jul 2025 14:09:03 +0800
Subject: [PATCH] WLANRTSYS-91362 iwlwav-hostapd-ng: UL SCS TC 4.47.1 issue fix

Fix up Commit: 408a725b4c94db0ca3bcc5482c3a88a7817c0dd2

Description: Post DL SCS merge, the UL wifi 7 4.47.1 cert TC cases failed.
Root Cause: The WIFI7 4.47.1 TC cases failed with invalid SCS response status. The frame was appended with 8extra bytes from the lower layers.
Fix: WA, 8 extra bytes were subtracted from the len in hostapd and minor error handling in scs remove path in driver.
Test: The cert TCs4.47.1 24G, 5G and 6G passed.

Signed-off-by: Karthik D A karthikda@maxlinear.com
---
 src/ap/mxl_wmm.c              | 3 ++-
 src/common/vendor_cmds_copy.h | 1 -
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ap/mxl_wmm.c b/src/ap/mxl_wmm.c
index 03f05465d..402080428 100644
--- a/src/ap/mxl_wmm.c
+++ b/src/ap/mxl_wmm.c
@@ -437,7 +437,8 @@ int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt
 		wpa_printf(MSG_ERROR, "SCS: mxl_scs_add_req - MLO not enabled/supported in station");
 		return WLAN_STATUS_UNSPECIFIED_FAILURE;
 	}
-
+	/* WLANRTSYS-91362: Workaround, Extra 8 bytes are appened to the frame from the lower layers */
+	len = len - 8;
 	end = (const u8 *) mgmt + len;
 	scs_add_req.aid = sta->aid;
 	scs_add_req.status = 0;
diff --git a/src/common/vendor_cmds_copy.h b/src/common/vendor_cmds_copy.h
index d37498cfc..7fbc65643 100644
--- a/src/common/vendor_cmds_copy.h
+++ b/src/common/vendor_cmds_copy.h
@@ -777,7 +777,6 @@ struct mxl_scs_add_req {
   u8  msduCountExponent;
   u16 mediumTime;
   u8 intraAccessPriority; //b0-b2 UP, b3= alternate queu, b4= drop eligibility, b5-b7 reserved
-  u8 version;
   u8 direction;
   struct mxl_scs_tclas_info tclasInfo;
   struct mxl_class_type4_params type4Params;
-- 
2.43.0

